modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/RocksDbTableStorageTest.java

```diff
@@ -206,4 +208,74 @@ void testRestart(
         assertThat(storage.getPartition(1), is(nullValue()));
         assertThat(storage.getPartition(0).read(testData), is(equalTo(testData)));
     }
+
+    /**
+     * Tests that restoring a snapshot clears all previous data.
+     */
+    @Test
+    void testRestoreSnapshot() {
+        PartitionStorage partitionStorage = storage.getOrCreatePartition(0);
+
+        var testData1 = new SimpleDataRow("foo".getBytes(StandardCharsets.UTF_8), "bar".getBytes(StandardCharsets.UTF_8));
+        var testData2 = new SimpleDataRow("baz".getBytes(StandardCharsets.UTF_8), "quux".getBytes(StandardCharsets.UTF_8));
+
+        Path snapshotDir = workDir.resolve("snapshot");
+
+        partitionStorage.write(testData1);
+
+        assertThat(partitionStorage.snapshot(snapshotDir), willBe(nullValue(Void.class)));
+
+        partitionStorage.write(testData2);
+
+        partitionStorage.restoreSnapshot(snapshotDir);
+
+        assertThat(partitionStorage.read(testData1), is(testData1));
+        assertThat(partitionStorage.read(testData2), is(nullValue()));
+    }
+
+    /**
+     * Tests that loading snapshots for one partition does not influence data in another.
+     */
+    @Test
+    void testSnapshotIndependence() {
```

Comment:
It looks like this test actually has three test scenarios:

1. When we restore one partition from a snapshot, other partitions are not affected
2. When we restore a partition from a snapshot, the data is reverted to the values that were in the snapshot (we don't see a key that was added after snapshot creation)
3. When we read by a row, actually its key gets used for addressing the value, and `value` part is ignored.

Item 3 does not seem to have anything with snapshotting, so my suggestion is to remove it (line 278).

As for items 1 and 2, I suggest to split the test scenario to two and test them independently.

Reply:
> Item 3 does not seem to have anything with snapshotting, so my suggestion is to remove it (line 278).

I don't agree, it checks that loading a snapshot for a different partition does not overwrite existing keys in another partition.

I'll split the test